src/libs/merge/kdbmerge.c

```diff
@@ -0,0 +1,527 @@
+
+#include "kdbmerge.h"
+#include "kdb.h"
+#include "kdbassert.h"
+#include "kdblogger.h"
+#include "kdbprivate.h"
+#include <ctype.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+int nonOverlapOnlyBaseCounter = 0;  // Conflict where only base is different (it exists) and our and their are empty.
+int nonOverlapBaseEmptyCounter = 0; // Conflict where only base is different (it is empty) and our=their (not empty).
+int nonOverlapAllExistCounter = 0;  // All three values exist, only base is different
+int overlap3different = 0;	  // Only overlap conflicts where all three keys exist and have different values
+// always access with getter
+int overlap1empty = 0; // counts overlaps where one set is empty. Is the double of the real amount in the end.
+
+int getNonOverlapAllExistConflicts (void)
```

Comment:
internal functions should be static. Please always add some comments to the functions.